lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -1531,22 +1543,29 @@ def run(self):
                 resolve=module.params['resolve_image']
             )
         except DockerException as e:
-            self.client.fail(
-                "Error looking for an image named %s: %s" % (image, e))
+            return self.client.fail(
```

Comment:
Why do you add `return` here (and below)? `self.client.fail()` terminates the program.

Reply:
I added it to stop static analysis tools from complaining that variables like `image_digest` and `current_service` may be referenced before assignment. I also feel that it makes it clearer that execution of that code will be stopped but when thinking a bit more it could also be confusing for a reader thinking that it will actually return and continue execution. You decide.

Reply:
Which static analyzer are you using for this? If it is a good one, it should be possible to tell it that that function will never return. Anyway, if we want to do something like this, I'd rather write `raise self.client.fail(...)` to make sure this is not interpreted as a usual return (and type checkers won't have trouble because it returns the wrong type).

Reply:
I use PyCharm so it's their internal tool. Just tried it with pylint and it does not warn about this so I'll remove the return.

Reply:
Ok. It would be nice if there's some way to tell PyCharm that fail never returns, that would be more elegant.